src/common/transformations/src/transformations/common_optimizations/compress_float_constants.cpp

```diff
@@ -25,19 +28,43 @@ std::shared_ptr<ov::Node> change_constant_precision_to_fp16(std::shared_ptr<ov::
     if (dst_data == nullptr)
         return nullptr;
 
-    bool is_overflow = false;
+    bool is_out_of_range = false;
+    int num_out_of_range = 0;
     for (size_t i = 0; i < size; ++i) {
-        if (src_data[i] > std::numeric_limits<ov::float16>::max()) {
+        // if it is smaller than the smallest positive normal fp16 but not zero
+        if (std::abs(src_data[i]) <= ov::float16::from_bits(0x0400) && src_data[i] != 0.0f) {
+            num_out_of_range++;
+            is_out_of_range = true;
+        } else if (src_data[i] > std::numeric_limits<ov::float16>::max()) {
             dst_data[i] = std::numeric_limits<ov::float16>::max();
-            is_overflow = true;
+            is_out_of_range = true;
+            num_out_of_range++;
         } else if (src_data[i] < std::numeric_limits<ov::float16>::lowest()) {
             dst_data[i] = std::numeric_limits<ov::float16>::lowest();
-            is_overflow = true;
+            is_out_of_range = true;
+            num_out_of_range++;
         } else {
             dst_data[i] = static_cast<ov::float16>(src_data[i]);
         }
     }
-    if (is_overflow) {
+
+    // thresholds in percents
+    // if more than 75% of a FP32 constant do not fit into FP16 keep in FP32
+    auto keep_threshold = static_cast<float>(ov::util::getenv_int("KEEP_FP32_THRESHOLD", 75));
```

Comment:
Looks like we are extending OV API by introducing this env var. What is the intended usage for it? How frequently does it make sense to change the threshold? Can we just fix it, and if a user doesn't see good accuracy, suggest to move to FP32 completely?

Reply:
At the moment, indeed this env variable is not used. Threshold 75% is ok for all cases we faced.

I used this `env` in the very beginning, when i was trying to find acceptable value. I found that values 60%, 70%, 90% are all equally good enough to keep critical constants in FP32, so i set it's default value to intermediate 75%.

Let's fix it to 75%. I removed specifying it from `env`.